pkg/controller/jobframework/reconciler.go

```diff
@@ -572,66 +576,91 @@ func (r *JobReconciler) recordAdmissionCheckUpdate(wl *kueue.Workload, job Gener
 	}
 }
 
-// IsAncestorJobManaged checks whether an ancestor job is managed by kueue.
-func (r *JobReconciler) IsAncestorJobManaged(ctx context.Context, jobObj client.Object) (bool, error) {
-	ancestor, err := r.getAncestorJobManagedByKueue(ctx, jobObj)
-	if err != nil {
-		return false, err
-	}
-	return ancestor != nil, nil
-}
-
 // getAncestorWorkload returns the Workload object of the Kueue-managed ancestor job.
-func (r *JobReconciler) getAncestorWorkload(ctx context.Context, jobObj client.Object) (*kueue.Workload, error) {
-	ancestor, err := r.getAncestorJobManagedByKueue(ctx, jobObj)
-	if err != nil || ancestor == nil {
-		return nil, err
-	}
+func (r *JobReconciler) getAncestorWorkload(ctx context.Context, ancestor client.Object) (*kueue.Workload, error) {
 	wlList := kueue.WorkloadList{}
 	if err := r.client.List(ctx, &wlList, client.InNamespace(ancestor.GetNamespace()), client.MatchingFields{indexer.OwnerReferenceUID: string(ancestor.GetUID())}); client.IgnoreNotFound(err) != nil {
 		return nil, err
 	}
 	if len(wlList.Items) > 0 {
 		// In theory the job can own multiple Workloads, we cannot do too much about it, maybe log it.
+		ctrl.LoggerFrom(ctx).V(2).Info(
+			"WARNING: The job has multiple associated Workloads.",
+			"job", ancestor.GetName(),
+			"workloads", klog.KObjSlice(wlList.Items),
+		)
 		return &wlList.Items[0], nil
 	}
 	return nil, nil
 }
 
-// getAncestorJobManagedByKueue traverses controllerRefs to find an ancestor job that is manged by Kueue (ie, it has a queue-name label).
-func (r *JobReconciler) getAncestorJobManagedByKueue(ctx context.Context, jobObj client.Object) (client.Object, error) {
+// GetAncestorJobManagedByKueue traverses controllerRefs to find the top-level ancestor Job managed by Kueue.
+// If manageJobsWithoutQueueName is set to false, it returns only Jobs with a queue-name.
+// If manageJobsWithoutQueueName is true, it may return a Job even if it doesn't have a queue-name.
+//
+// Examples:
+//
+// With manageJobsWithoutQueueName=true:
+// Job -> JobSet -> AppWrapper => nil
+// Job (queue-name) -> JobSet (queue-name) -> AppWrapper => JobSet
+// Job (queue-name) -> JobSet -> AppWrapper (queue-name) => AppWrapper
+// Job (queue-name) -> JobSet (queue-name) -> AppWrapper (queue-name) => AppWrapper
+// Job -> JobSet (disabled) -> AppWrapper (queue-name) => AppWrapper
+//
+// With manageJobsWithoutQueueName=false:
```

Comment:
```suggestion
// With manageJobsWithoutQueueName=true:
```
IIUC the examples below

Reply:
Ah, my bad. Thanks!